### PR TITLE
TMA-1190: Add test env clean-up script

### PR DESCRIPTION
**Pull request by panjan**

### CODE REVIEW

File: bin/test_projects_cleanup.rb [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      project.add && project.add.output_stage && project.add.output_stage.delete
+      project.delete
+    else
+      puts "Would delete: #{project.pid} - #{project.title} - #{project.created}"
+    end
+  end
+  puts "#{filtered_projects.length} projects matching \"#{title}\" #{'would be ' unless force}deleted."
+end
+
+def delete_ads_by_title(title, client, days = 14, force = false)
```
Comment (yenkins) [inline bin/test_projects_cleanup.rb]: info: Has boolean parameter 'force' (BooleanParameter) (reek)

Comment (yenkins) [inline bin/test_projects_cleanup.rb]: info: Has 4 parameters (LongParameterList) (reek)

Comment (yenkins) [inline bin/test_projects_cleanup.rb]: info: Has approx 15 statements (TooManyStatements) (reek)

File: bin/test_projects_cleanup.rb [hunk 2]
```diff
@@ -0,0 +1,78 @@
+example = 'login=rubydev+admin@gooddata.com password=secret server=https://staging2-lcm-prod.intgdc.com ruby delete_projects.rb'
+if ARGV.include?('-h')
+  puts 'Example:'
+  puts example
+  exit 0
+end
+
+require 'gooddata'
+
+def delete_project_by_title(title, projects, days = 14, force = false)
```
Comment (yenkins) [inline bin/test_projects_cleanup.rb]: info: Has boolean parameter 'force' (BooleanParameter) (reek)

Comment (yenkins) [inline bin/test_projects_cleanup.rb]: info: Has 4 parameters (LongParameterList) (reek)

Comment (yenkins) [inline bin/test_projects_cleanup.rb]: info: Has approx 9 statements (TooManyStatements) (reek)

File: bin/test_projects_cleanup.rb [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  warehouses = client.warehouses
+  return if warehouses.empty?
+  deleted = 0
+  warehouses.each do |warehouse|
+    next unless warehouse.title.match(title)
+    dead_line = Time.now - days * 60 * 60 * 24
+    created = Time.parse(warehouse.data["created"])
+    next if created > dead_line
+    begin
+      if force
```
Comment (yenkins) [inline bin/test_projects_cleanup.rb]: info: Is controlled by argument 'force' (ControlParameter) (reek)

File: bin/test_projects_cleanup.rb [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+require 'gooddata'
+
+def delete_project_by_title(title, projects, days = 14, force = false)
+  dead_line = Time.now - days * 60 * 60 * 24
+  filtered_projects = projects.select do |p|
+    p.title.match(title) && p.created < dead_line
+  end
+  filtered_projects.each do |project|
+    if force
```
Comment (yenkins) [inline bin/test_projects_cleanup.rb]: info: Is controlled by argument 'force' (ControlParameter) (reek)

File: bin/test_projects_cleanup.rb [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+  end
+  puts "#{filtered_projects.length} projects matching \"#{title}\" #{'would be ' unless force}deleted."
+end
+
+def delete_ads_by_title(title, client, days = 14, force = false)
+  warehouses = client.warehouses
+  return if warehouses.empty?
+  deleted = 0
+  warehouses.each do |warehouse|
+    next unless warehouse.title.match(title)
```
Comment (yenkins) [inline bin/test_projects_cleanup.rb]: info: Calls 'warehouse.title' 4 times (DuplicateMethodCall) (reek)

Comment (yenkins) [inline bin/test_projects_cleanup.rb]: info: Refers to 'warehouse' more than self (maybe move it to another class?) (FeatureEnvy) (reek)

Comment (yenkins) [inline bin/test_projects_cleanup.rb]: warning: Layout/EmptyLineAfterGuardClause: Add empty line after guard clause. (rubocop)

File: bin/test_projects_cleanup.rb [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+def delete_project_by_title(title, projects, days = 14, force = false)
+  dead_line = Time.now - days * 60 * 60 * 24
+  filtered_projects = projects.select do |p|
+    p.title.match(title) && p.created < dead_line
+  end
+  filtered_projects.each do |project|
+    if force
+      puts "Deleting: #{project.pid} - #{project.title} - #{project.created}"
+      project.add && project.add.output_stage && project.add.output_stage.delete
```
Comment (yenkins) [inline bin/test_projects_cleanup.rb]: info: Calls 'project.add' 3 times (DuplicateMethodCall) (reek)

Comment (yenkins) [inline bin/test_projects_cleanup.rb]: info: Calls 'project.add.output_stage' 2 times (DuplicateMethodCall) (reek)

File: bin/test_projects_cleanup.rb [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+require 'gooddata'
+
+def delete_project_by_title(title, projects, days = 14, force = false)
+  dead_line = Time.now - days * 60 * 60 * 24
+  filtered_projects = projects.select do |p|
+    p.title.match(title) && p.created < dead_line
+  end
+  filtered_projects.each do |project|
+    if force
+      puts "Deleting: #{project.pid} - #{project.title} - #{project.created}"
```
Comment (yenkins) [inline bin/test_projects_cleanup.rb]: info: Calls 'project.created' 2 times (DuplicateMethodCall) (reek)

Comment (yenkins) [inline bin/test_projects_cleanup.rb]: info: Calls 'project.pid' 2 times (DuplicateMethodCall) (reek)

Comment (yenkins) [inline bin/test_projects_cleanup.rb]: info: Calls 'project.title' 2 times (DuplicateMethodCall) (reek)

Comment (yenkins) [inline bin/test_projects_cleanup.rb]: info: Refers to 'project' more than self (maybe move it to another class?) (FeatureEnvy) (reek)

File: bin/test_projects_cleanup.rb [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+      end
+      deleted += 1
+    rescue => e
+      puts "Failed to delete #{warehouse.title}: #{e}"
+    end
+  end
+  puts "#{deleted} ADS instances with title \"#{title}\" #{'would be ' unless force}deleted."
+end
+
+def clean_up!(client, force, days)
```
Comment (yenkins) [inline bin/test_projects_cleanup.rb]: info: Has approx 13 statements (TooManyStatements) (reek)

File: bin/test_projects_cleanup.rb [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+    next if created > dead_line
+    begin
+      if force
+        puts "Deleting: #{warehouse.title} - #{created}"
+        warehouse.delete
+      else
+        puts "Would delete:  #{warehouse.title} - #{created}"
+      end
+      deleted += 1
+    rescue => e
```
Comment (yenkins) [inline bin/test_projects_cleanup.rb]: info: Has the variable name 'e' (UncommunicativeVariableName) (reek)

Comment (yenkins) [inline bin/test_projects_cleanup.rb]: warning: Style/RescueStandardError: Avoid rescuing without specifying an error class. (rubocop)

File: bin/test_projects_cleanup.rb [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+  puts 'Example:'
+  puts example
+  exit 0
+end
+
+require 'gooddata'
+
+def delete_project_by_title(title, projects, days = 14, force = false)
+  dead_line = Time.now - days * 60 * 60 * 24
+  filtered_projects = projects.select do |p|
```
Comment (yenkins) [inline bin/test_projects_cleanup.rb]: info: Has the variable name 'p' (UncommunicativeVariableName) (reek)

File: bin/test_projects_cleanup.rb [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+    else
+      puts "Would delete: #{project.pid} - #{project.title} - #{project.created}"
+    end
+  end
+  puts "#{filtered_projects.length} projects matching \"#{title}\" #{'would be ' unless force}deleted."
+end
+
+def delete_ads_by_title(title, client, days = 14, force = false)
+  warehouses = client.warehouses
+  return if warehouses.empty?
```
Comment (yenkins) [inline bin/test_projects_cleanup.rb]: warning: Layout/EmptyLineAfterGuardClause: Add empty line after guard clause. (rubocop)

File: bin/test_projects_cleanup.rb [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+  deleted = 0
+  warehouses.each do |warehouse|
+    next unless warehouse.title.match(title)
+    dead_line = Time.now - days * 60 * 60 * 24
+    created = Time.parse(warehouse.data["created"])
+    next if created > dead_line
```
Comment (yenkins) [inline bin/test_projects_cleanup.rb]: warning: Layout/EmptyLineAfterGuardClause: Add empty line after guard clause. (rubocop)

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: bin/test_projects_cleanup.rb [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+      project_add && project_add.output_stage && project_add.output_stage.delete
+      project.delete
+    else
+      puts "Would delete: #{project.pid} - #{project.title} - #{project.created}"
+    end
+  end
+  puts "#{filtered_projects.length} projects matching \"#{title}\" #{'would be ' unless force}deleted."
+end
+
+def delete_ads_by_title(title, client, days = 14, force = false)
```
Comment (yenkins) [inline bin/test_projects_cleanup.rb]: info: Has approx 16 statements (TooManyStatements) (reek)

File: bin/test_projects_cleanup.rb [hunk 14]
```diff
@@ -0,0 +1,83 @@
+example = 'login=rubydev+admin@gooddata.com password=secret server=https://staging2-lcm-prod.intgdc.com ruby delete_projects.rb'
+if ARGV.include?('-h')
+  puts 'Example:'
+  puts example
+  exit 0
+end
+
+require 'gooddata'
+
+def delete_project_by_title(title, projects, days = 14, force = false)
```
Comment (yenkins) [inline bin/test_projects_cleanup.rb]: info: Has approx 10 statements (TooManyStatements) (reek)

File: bin/test_projects_cleanup.rb [hunk 15]
```diff
@@ -0,0 +1,83 @@
...
+  dead_line = Time.now - days * 60 * 60 * 24
+  filtered_projects = projects.select do |p|
+    p.title.match(title) && p.created < dead_line
+  end
+  filtered_projects.each do |project|
+    if force
+      puts "Deleting: #{project.pid} - #{project.title} - #{project.created}"
+      project_add = project.add
+      project_add && project_add.output_stage && project_add.output_stage.delete
```
Comment (yenkins) [inline bin/test_projects_cleanup.rb]: info: Calls 'project_add.output_stage' 2 times (DuplicateMethodCall) (reek)

File: bin/test_projects_cleanup.rb [hunk 16]
```diff
@@ -0,0 +1,83 @@
...
+
+    begin
+      if force
+        puts "Deleting: #{warehouse_title} - #{created}"
+        warehouse.delete
+      else
+        puts "Would delete:  #{warehouse_title} - #{created}"
+      end
+      deleted += 1
+    rescue StandardError => e
```
Comment (yenkins) [inline bin/test_projects_cleanup.rb]: info: Has the variable name 'e' (UncommunicativeVariableName) (reek)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/855/ : SUCCESS in 11s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1668/ : SUCCESS in 1m 39s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1929/ : SUCCESS in 1m 34s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/106/ : SUCCESS in 8m 22s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: bin/test_projects_cleanup.rb [hunk 17]
```diff
@@ -0,0 +1,97 @@
...
+  delete_project_by_title(/lcm-test-fixture/, projects, days, force)
+  delete_project_by_title(/Test MASTER project/, projects, days, force)
+  delete_project_by_title(/Test MINOR project/, projects, days, force)
+  delete_project_by_title(/^Test project$/, projects, days, force)
+  delete_project_by_title(/userprov-e2e-testing/, projects, days, force)
+  delete_ads_by_title(/Development ADS/, client, days, force)
+  delete_ads_by_title(/Production ADS/, client, days, force)
+end
+
+def init_client(username, password, server)
```
Comment (yenkins) [inline bin/test_projects_cleanup.rb]: info: Doesn't depend on instance state (maybe move it to another class?) (UtilityFunction) (reek)

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/857/ : SUCCESS in 6s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1670/ : SUCCESS in 2m 58s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1931/ : FAILURE in 41s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/108/ : SUCCESS in 8m 28s

**Pull request closed without merging**